src/evaluation/zero_shot.py

```diff
@@ -20,10 +20,18 @@ def __init__(self,
         self.device = "cuda" if torch.cuda.is_available() else "cpu"
         self.model, _ = clip.load("ViT-B/32", device=self.device)
 
-        labels_ = [prompt_func(l) for l in self.labels] # prompt engineering
-        with torch.no_grad():
-            self.lab_embeds = self.model.encode_text(clip.tokenize(labels_).to(self.device))
-            self.lab_embeds /= self.lab_embeds.norm(dim=-1, keepdim=True)
+        #labels_ = [prompt_func(l) for l in self.labels] # prompt engineering
+        
+        
+        embed_list = []
+        for template in templates:#Done as a loop over lists because model can't handle 700 x 30
+            texts = [template.format(classname) for classname in self.labels]
+            with torch.no_grad():
+                embed_list.append(self.model.encode_text(clip.tokenize(texts).to(self.device)))
+        
+        self.lab_embeds = torch.stack(embed_list, dim = 0)
+        self.lab_embeds /= self.lab_embeds.norm(dim=-1, keepdim=True)
+        self.lab_embeds = self.lab_embeds.to(torch.float32)
```

Comment:
is this necessary?

Reply:
While there's probably a more elegant way to do, the answer is probably yes. The other way is to create an empty tensor with the right shape ahead of time and populate it, as far as I'm aware.

Reply:
I mean the .to(torch.float32)

Reply:
Ah. Sorry, that might be an OS related thing, it was needed to get Windows working.. Testing without on other OS would be useful though.